app/Libraries/Groups.php

```diff
@@ -6,41 +6,35 @@
 namespace App\Libraries;
 
 use App\Models\Group;
-use App\Traits\Memoizes;
+use App\Traits\LocallyCached;
 use Illuminate\Database\Eloquent\Collection;
 
 class Groups
 {
-    use Memoizes;
+    use LocallyCached;
 
     /**
      * Get all groups.
      */
     public function all(): Collection
     {
-        return $this->memoize(__FUNCTION__, function () {
-            return $this->fetch();
-        });
+        return $this->memoize(__FUNCTION__, fn () => $this->cachedFetch());
```

Comment:
I can't help but think this should be a different function that instead that accepts what would go in `fetch()` as the callable instead of `fetch()` being an abstract method.

`cachedFetch()` seems to exist just to call `fetch()` (which needs to be implemented) and be memoized